language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/hover/HoverUtil.java

```diff
@@ -81,6 +87,16 @@ public static Hover getHover(HoverContext context) {
         // Check for the cancellation after the time consuming operation
         context.checkCancelled();
         if (symbolAtCursor.isEmpty()) {
+            Range nodeRange = new Range(context.getCursorPosition(), context.getCursorPosition());
+            NonTerminalNode nodeAtCursor = CommonUtil.findNode(nodeRange, srcFile.get().syntaxTree());
+            if (nodeAtCursor != null) {
+                MatchedExpressionNodeResolver expressionResolver = new MatchedExpressionNodeResolver(nodeAtCursor);
```

Comment:
I see that we are reusing the Code Action's node finder util here. Shall we make a note to introduce a common API/ visitor in the future?

Reply:
I think this visitor is useful for us for use cases like this. Rather than checking each expression kind (via syntax kind), this is clean. Is it ok if we use it everywhere?

Reply:
The best option would be now, to take this out to a common place. Please create a separate task to refactor this.